samples/net/mqtt_publisher/src/main.c

```diff
@@ -243,45 +269,38 @@ static void publisher(void)
 {
 	int i, rc;
 
-	/* The net_ctx variable must be ready BEFORE passing it to the MQTT API.
-	 */
-	for (i = 0; i < CONN_TRIES; i++) {
-		rc = network_setup(&net_ctx, ZEPHYR_ADDR, SERVER_ADDR,
-				   SERVER_PORT);
-		if (!rc) {
-			goto connected;
-		}
-	}
-
-	PRINT_RESULT("network_setup", rc);
-	goto exit_app;
-
-connected:
 	/* Set everything to 0 and later just assign the required fields. */
 	memset(&client_ctx, 0x00, sizeof(client_ctx));
 
-	/* The network context is the only field that must be set BEFORE
-	 * calling the mqtt_init routine.
-	 */
-	client_ctx.mqtt_ctx.net_ctx = net_ctx;
-
 	/* connect, disconnect and malformed may be set to NULL */
 	client_ctx.mqtt_ctx.connect = connect_cb;
 
 	client_ctx.mqtt_ctx.disconnect = disconnect_cb;
 	client_ctx.mqtt_ctx.malformed = malformed_cb;
 
+	client_ctx.mqtt_ctx.net_app_init_timeout = APP_NET_INIT_TIMEOUT;
 	client_ctx.mqtt_ctx.net_timeout = APP_TX_RX_TIMEOUT;
 
+	client_ctx.mqtt_ctx.peer_addr_str = SERVER_ADDR;
+	client_ctx.mqtt_ctx.peer_port = SERVER_PORT;
+
+#if defined(CONFIG_MQTT_LIB_TLS)
+	/** TLS setup */
+	client_ctx.mqtt_ctx.request_buf = tls_request_buf;
+	client_ctx.mqtt_ctx.request_buf_len = TLS_REQUEST_BUF_SIZE;
+	client_ctx.mqtt_ctx.personalization_data = TLS_PRIVATE_DATA;
+	client_ctx.mqtt_ctx.personalization_data_len = strlen(TLS_PRIVATE_DATA);
+	client_ctx.mqtt_ctx.cert_host = TLS_SNI_HOSTNAME;
+	client_ctx.mqtt_ctx.tls_mem_pool = &tls_mem_pool;
+	client_ctx.mqtt_ctx.tls_stack = tls_stack;
+	client_ctx.mqtt_ctx.tls_stack_size = CONFIG_NET_APP_TLS_STACK_SIZE;
```

Comment:
You need to use K_THREAD_STACK_SIZEOF() macro here like this

...tls_stack_size = K_THREAD_STACK_SIZEOF(tls_stack);